src/controller/physicaldevice/ConnectedPhysicalDeviceManager.cpp

```diff
@@ -0,0 +1,45 @@
+#include "ConnectedPhysicalDeviceManager.h"
+
+namespace Ship {
+ConnectedPhysicalDeviceManager::ConnectedPhysicalDeviceManager() {
+}
+
+ConnectedPhysicalDeviceManager::~ConnectedPhysicalDeviceManager() {
+}
+
+std::unordered_map<int32_t, SDL_GameController*>
+ConnectedPhysicalDeviceManager::GetConnectedSDLGamepadsForPort(uint8_t portIndex) {
+    // todo: filter somehow
+    return mConnectedSDLGamepads;
+}
```

Comment:
this is the big todo - now that mappings aren't tied to devices this means that if you have mappings on all the ports every controller will send inputs to every port

i'd like to find a way to handle this without introducing the type of complexity that led to the issues we have with the pre-simplified system

Reply:
calling this one good as of https://github.com/Kenix3/libultraship/commit/96e7d20d44d8013983d3633b67aa0189793d90ca